Mathlib/Analysis/InnerProductSpace/Calculus.lean

```diff
@@ -245,6 +245,40 @@ theorem DifferentiableAt.norm (hf : DifferentiableAt ℝ f x) (h0 : f x ≠ 0) :
   ((contDiffAt_norm 𝕜 h0).differentiableAt le_rfl).comp x hf
 #align differentiable_at.norm DifferentiableAt.norm
 
+theorem not_differentiableAt_abs_zero : ¬ DifferentiableAt ℝ (fun x : ℝ ↦ abs x) 0 := by {
+  rewrite [DifferentiableAt]
+  push_neg
+  intro f
+  simp [HasFDerivAt, HasFDerivAtFilter, Asymptotics.isLittleO_iff]
+  use 2⁻¹
+  norm_num
+  rewrite [Filter.frequently_iff]
+  intro U hU
+  obtain ⟨ε, hε⟩ : ∃ ε ∈ U, ε ≠ 0 ∧ f ε ≤ 0 := by {
+    obtain ⟨a, b, hab0, habU⟩ := mem_nhds_iff_exists_Ioo_subset.mp hU
+    obtain ⟨ha0, hb0⟩ := Set.mem_Ioo.mp hab0
+    obtain ⟨a', ha'⟩ := Set.nonempty_Ioo.mpr ha0
+    obtain ⟨b', hb'⟩ := Set.nonempty_Ioo.mpr hb0
+    by_cases hf1 : f 1 ≥ 0
+    · use a', habU (Set.Ioo_subset_Ioo_right (le_of_lt hb0) ha'),
+        ne_of_lt (Set.mem_Ioo.mp ha').right
+      rewrite [← mul_one a', ← smul_eq_mul, ContinuousLinearMap.map_smul, @smul_eq_mul _ _ _ (f 1),
+        ← mul_zero a', mul_le_mul_left_of_neg (Set.mem_Ioo.mp ha').right]
+      exact hf1
+    · use b', habU (Set.Ioo_subset_Ioo_left (le_of_lt ha0) hb'), ne_of_gt (Set.mem_Ioo.mp hb').left
+      push_neg at hf1
+      rewrite [← mul_one b', ← smul_eq_mul, ContinuousLinearMap.map_smul, @smul_eq_mul _ _ _ (f 1),
```

Comment:
```suggestion
      rw [← mul_one b', ← smul_eq_mul, ContinuousLinearMap.map_smul, @smul_eq_mul _ _ _ (f 1),
```
Using rw is a lot more common than rewrite, so I'd recommend getting in to the habit of doing that also

Reply:
Actually I recently got rid of the habit of using `rw` from Lean 3 era and switched to using `rewrite` because apparently they are marginally cheaper in Lean 4? Maybe no one else does it this way...

Reply:
Well I think it was marginally cheaper in lean 3 also I believe. I think its fine to keep it as `rewrite` but just wanted to make sure you are aware

Reply:
Of course. Thanks for the review!